metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -225,6 +225,7 @@ def create(
         obj.echo_always(str(flow), err=False, no_bold=True)
         # TODO: Support echo-ing Argo Events Sensor template
     else:
+        flow.cleanup_previous()
```

Comment:
I believe there is an issue with running the cleanup only as part of the deploy command. Consider the following scenario:

- user has deployed a sensor flow to namespace `dev-A`, sensor exists in the same namespace
- this feature gets released and the Argo sensor namespace env_var gets configured to `dev-B`
- the user decides that they want to get rid of their flow, and issue an `argo-workflows delete` directly.

Wouldn't this only check for the currently configured sensor namespace, leaving the original sensor in `dev-A` untouched?

Reply:
At least in the case of `delete` it seems sensible to me to keep the concern of cleaning up resources as part of the method.

Reply:
For the legacy case, we need to do cleanup before creating a new argo workflow.  This can only happen if we clean up before calling `flow.deploy()`.

One reason I did this is that there is some inconsistency in the API.  The current implementation already does some sort of "cleanup".  For schedules, we create/suspend them within the `schedule_workflow_template` method which is in the `argo_client.py`, which is called within the `schedule` method in `argo_workflows.py`.  However the sensors are created/deleted within the `schedule` method in `argo_workflows.py`.  So I wanted to move this all into one place, but I couldn't put my code into this `schedule` method.  The previous annotations for the triggers would be overwritten before the `schedule`method.

Currently, the `flow.schedule` function is used to create both the schedule and the sensor with the assumption that the sensor will get overwritten because it always exists in the same namespace.  So in your scenario, if you configured you local environment with `dev-B` then you would have to do an `argo-workflows create` in order to create a sensor in `dev-B`.  Once you did that, it would delete the sensor in `dev-A` due to this line:

https://github.com/dhpollack/metaflow/blob/feature/configure-sensor-namespace/metaflow/plugins/argo/argo_workflows.py#L195